samza-kafka/src/main/scala/org/apache/samza/checkpoint/kafka/KafkaCheckpointManager.scala

```diff
@@ -64,14 +64,14 @@ class KafkaCheckpointManager(
                               checkpointTopicProperties: Properties = new Properties) extends CheckpointManager with Logging {
 
   var taskNames = Set[TaskName]()
-  @volatile var  systemProducer: SystemProducer = null
+  @volatile var systemProducer: SystemProducer = null
+  var systemConsumer: SystemConsumer = null
```

Comment:
Maybe instead of the comment on line 120 say something like "This class is thread safe for writing but not for reading checkpoints. This is currently OK since checkpoints are only on the main thread."